### ISSUE TITLE
Nominatim::_construct expects min 3 arguments; config & factory only allow for 2

### ISSUE THREAD

**Issue by kshishkin** [geocoder-php/BazingaGeocoderBundle#195]
When trying to use Nominatim provider I get this error:
Too few arguments to function Geocoder\Provider\Nominatim\Nominatim::__construct(), 2 passed

This happens here:
```php
//NominatimFactory.php:30
return new Nominatim($httplug, $config['root_url']);
```
Nominatim since version 4.1 requires $userAgent and allows for $referer:

[(v.4.1 commit line)](https://github.com/geocoder-php/nominatim-provider/compare/4.1.0...master#diff-21af159c905d0493a0e92eed8ee6937bR68)

```php
//Nominatim.php:68
-    public function __construct(HttpClient $client, $rootUrl)
+    public function __construct(HttpClient $client, $rootUrl, string $userAgent, string $referer = ''
```
my config: 
```yml
bazinga_geocoder:
    providers:
        nominatim:
            factory: Bazinga\GeocoderBundle\ProviderFactory\NominatimFactory
            options:
                httplug_client: '@httplug.client' # When using HTTPlugBundle
                root_url: 'https://nominatim.openstreetmap.org'
```

**Comment by toooni** [comment]
This Issue can be closed now.

**Comment by Nyholm** [comment]
Closed by #198
